### PR TITLE
Document how to override configuration by type

### LINKED ISSUE #240: Document how Domain Config can override any type of configuration
Domain Config can be used to override any type of configuration, but:
* The README file and test module overrides are limited to simple configuration.
* The steps to override configuration of a type other than simple configuration are non-intuitive.

### PR DESCRIPTION
**Pull request by nedjo**
Fixes #240.

### CODE REVIEW

File: domain_config/README.md [hunk 1]
```diff
@@ -1,24 +1,51 @@
...
+admin_compact_mode: false
+weight_select_max: 100
+langcode: en
+default_langcode: en
+```
+
+Edit the file to change the site name, as follows:
 
 ```YAML
 uuid: 536db351-7aec-407e-a746-70ba9cc9f190
```
Comment (agentrickard) [inline domain_config/README.md]: I believe we also have to remove the UUID, else it causes import problems.

**Comment by agentrickard** [comment]
The failing test herre is a know inconsistency with Travis and can be ignored.

File: domain_config/README.md (hunk 1 above)
Comment (zerolab) [inline domain_config/README.md]: Most definitely! See #229

Comment (agentrickard) [inline domain_config/README.md]: Let's put a big, fat warning about that in the README.

File: domain_config/README.md [hunk 2]
```diff
@@ -1,28 +1,38 @@
...
+for a single export:
+`admin/config/development/configuration/single/export`
+
+In this case, the type of configuration to be exported is "Simple
+configuration" and the particular item is named `system.site`.
+
+You should have a file like the following:
 
 ```YAML
 uuid: 536db351-7aec-407e-a746-70ba9cc9f190
```
Comment (agentrickard) [inline domain_config/README.md:34]: I believe we specifically have to remove the UUID line or the import fails.

Comment (zerolab) [inline domain_config/README.md:34]: In nedjo's defence, a bit further it does state "An override file should contain only the specific data that you want to override."

Best to be very explicit about the need to exclude the UUID line.

**Pull request merged**